tests/unit/test_cli.py

```diff
@@ -577,37 +580,72 @@ def test_invalid_flags(self, cli_args: list[str], expect_error: str) -> None:
         assert_pattern_in_output(expect_error, result.output)
 
     @pytest.mark.parametrize(
-        "request_output",
+        "request_output,invoke_extra_args,expected_validation_class",
         [
-            RequestOutput.empty(),
-            RequestOutput.from_obj_list(
-                components=[
-                    Component(
-                        name="cool-package",
-                        version="v1.0.0",
-                        purl="pkg:generic/cool-package@v1.0.0",
-                        type="library",
-                    )
-                ],
-                environment_variables=[
-                    EnvironmentVariable(name="GOMOD_SOMETHING", value="yes"),
-                ],
-                project_files=[],
+            (RequestOutput.empty(), [], Sbom),
+            (
+                RequestOutput.from_obj_list(
+                    components=[
+                        Component(
+                            name="cool-package",
+                            version="v1.0.0",
+                            purl="pkg:generic/cool-package@v1.0.0",
+                            type="library",
+                        )
+                    ],
+                    environment_variables=[
+                        EnvironmentVariable(name="GOMOD_SOMETHING", value="yes"),
+                    ],
+                    project_files=[],
+                ),
+                [],
+                Sbom,
+            ),
+            (RequestOutput.empty(), ["--sbom-type", "spdx"], SPDXSbom),
+            (
+                RequestOutput.from_obj_list(
+                    components=[
+                        Component(
+                            name="cool-package",
+                            version="v1.0.0",
+                            purl="pkg:generic/cool-package@v1.0.0",
+                            type="library",
+                        )
+                    ],
+                    environment_variables=[
+                        EnvironmentVariable(name="GOMOD_SOMETHING", value="yes"),
+                    ],
+                    project_files=[],
+                ),
+                ["--sbom-type", "spdx"],
+                SPDXSbom,
             ),
         ],
     )
-    def test_write_json_output(self, request_output: RequestOutput, tmp_cwd: Path) -> None:
+    def test_write_json_output(
```

Comment:
Could you please keep the original test intact and add another test case to ensure that the new functionality also works? This would separate a new feature from existing ones and would help insure backwards compatibility.